support/sier-codec-js/src/lib.rs

```diff
@@ -0,0 +1,54 @@
+use sier_codec::Parser;
+use wasm_bindgen::prelude::*;
+//TODO(melatron): Create a JS Class that holds the file_defs inside and
+// serialize/deserialize are methods of this class.
+#[wasm_bindgen]
+pub fn serialize(js_object: JsValue, file_defs: &str, struct_def: &str) -> js_sys::Uint8Array {
+    if !js_object.is_object() {
+        wasm_bindgen::throw_str("Provided argument is not a JS Object.");
+    }
+
+    let json = js_sys::JSON::stringify(&js_object).unwrap_or_else(|e| {
+        wasm_bindgen::throw_val(e);
+    });
+    let json = json
+        .as_string()
+        .expect("Parsed as string from JSON::stringify");
+
+    let mut parser = Parser::default();
+    parser.add_file_defs(file_defs).unwrap_or_else(|e| {
+        wasm_bindgen::throw_str(format!("Wrong file definitions provided {0}", e).as_str());
```

Comment:
Errors here aren't necessarily "Wrong file definitions". I would just pass through the error (possibly stringified).